yo/app/scripts/directives/select-records.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (C) 2013 – 2015  SLUB Dresden & Avantgarde Labs GmbH (<code@dswarm.org>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+angular.module('dmpApp')
+    .controller('SelectRecordsCtrl', function($scope, $http, $q, $modal, $modalInstance, loDash, gdmParser, schemaParser, filterHelper, PubSub, DataModelResource, project, recordsSelected) {
+
+        $scope.internalName = 'Selected Records Widget';
+
+        $scope.activeFilterObject = project;
+        $scope.recordsSelected = recordsSelected;
+        $scope.filters = [];
+
+        $scope.dataSources = [];
+        $scope.dataSchema = angular.copy(project.input_data_model.schema);
+        $scope.dataLoaded = false;
+        var isRemoveFilter = false;
+        var isResetRecordSelection = false;
+        var isRecordsHaveBeenSearched = true;
+        $scope.filterSelectorShown = false;
+
+        var rawDataSources = project._$selectedRecords;
+        var inputDataModel = project.input_data_model;
+        $scope.model = {
+            selectedRecords: rawDataSources,
+            inputDataModel: inputDataModel
+        };
+        var inputFilterCollection = [];
+
+        // deactivated until further notice
+        /* jshint ignore:start */
+        function restrictSchema(schema, pathId) {
```

Comment:
It's bad enough that we have unused method dragging around in one place, we certainly don't need a still unused copy/paste of this :smile: